lib/siwapp/commons/meta_attribute.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Siwapp.Commons.MetaAttribute do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  embedded_schema do
+    field :temp_id, :string, virtual: true
+    #   field :delete, :boolean, virtual: true
+    field :key, :string
```

Comment:
For this pattern with a set of unique keys, you could try setting `:key` as the primary key of the embedded schema (I'm almost sure any binary can be a primary key, not only UUIDs). This way you would get rid of the `:temp_id` field. https://hexdocs.pm/ecto/Ecto.Schema.html#module-primary-keys

Reply:
forget this @nurietzsche2021